libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -796,7 +798,6 @@ string YulUtilFunctions::arrayConvertLengthToSize(ArrayType const& _type)
 }
 string YulUtilFunctions::arrayAllocationSizeFunction(ArrayType const& _type)
 {
-	solAssert(_type.dataStoredIn(DataLocation::Memory), "");
```

Comment:
This assert kind of makes sense here.

Reply:
It fails for arrays that are members of structs, which have `storage` set there

Reply:
But then you should change the storage location to memory instead.

Reply:
Yeah, I think it is safer to keep the assertion here and change the struct zero function instead.

Reply:
Ok!